packages/pwa-kit-react-sdk/src/ssr/universal/components/storefront-preview/utils.js

```diff
@@ -39,6 +39,11 @@ export const detectStorefrontPreview = () => {
     return Boolean(parentOrigin) && isParentOriginTrusted(parentOrigin)
 }
 
+export const detectClientOnlyQueryParam = () => {
+    const queryParams = new URLSearchParams(window.location.search)
```

Comment:
```suggestion
    if (typeof window === 'undefined') return false
    const queryParams = new URLSearchParams(window.location.search)
```
This is a `universal` component, so we should be sure to guard server-side usage.

Also, this method isn't specific to Storefront Preview any more - maybe move it to `universal/utils.js`?

Reply:
Alternatively, it's a super small function - might be simpler to just put the logic directly in `main.jsx`.

Reply:
Thanks Will, I moved the logic to `main.jsx`